### PR TITLE
feat(free-trials): Free trials example for Dashboard Pages

### PR DESCRIPTION
**Pull request by denysdovhan**
This PR implements the [Free Trials](https://dev.wix.com/docs/build-apps/launch-your-app/pricing-and-billing/set-up-and-manage-free-trials) example for dashboard pages template.

### CODE REVIEW

File: custom-products-catalog/template/src/dashboard/pages/page.tsx [hunk 1]
```diff
@@ -160,10 +163,13 @@ function Products() {
...
-          <PrimaryPageButton
-            text="Add Product"
-            onClick={() => setShown(!shown)}
-          />
+          <PremiumFeature>
+            <PrimaryPageButton
+              disabled={isFreeApp}
+              text="Add Product"
+              onClick={() => setShown(!shown)}
+            />
```
Comment (AdirAmsalem) [inline custom-products-catalog/template/src/dashboard/pages/page.tsx:171]: This is not related to your changes, so let's do this in a separate PR (if it's small):

`PrimaryPageButton` seem deprecated, we should update it with the suggested alternative.

image

File: custom-products-catalog/template/src/dashboard/components/premium-feature.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export function PremiumFeature({ children }: PremiumFeatureProps) {
+    const isFreeApp = useIsFreeApp();
+    const isFreeTrialAvailable = useFreeTrialAvailable();
+    const navigateToPricingPage = useNavigateToPricingPage();
+
+    if (isFreeApp && isFreeTrialAvailable) {
+      return (
+        <FloatingHelper
+          placement="bottom-end"
+          appearance='light'
```
Comment (AdirAmsalem) [inline custom-products-catalog/template/src/dashboard/components/premium-feature.tsx:18]: let's be consistent with single quotes? (in other places too)

**Review by AdirAmsalem — COMMENTED** [review]
Small stuff, nice work 👍 

p.s.
Many of the comments also apply to https://github.com/wix/cli-app-templates/pull/58 (e.g., multiple server calls, quote consistency, and the ‘permanent paywall’ scenario). I’ll review it only after we make the necessary adjustments.

File: custom-products-catalog/template/src/dashboard/pages/page.tsx [hunk 3]
```diff
@@ -160,10 +163,13 @@ function Products() {
...
+          </PremiumFeature>
```
Comment (AdirAmsalem) [inline custom-products-catalog/template/src/dashboard/pages/page.tsx:172]: What do you think about including the `isFree && !isFreeTrialAvailable` case as well? To demonstrate the scenario where users have already used their free trial, are now paywalled, and need to upgrade to add new products.

File: custom-products-catalog/template/src/dashboard/components/premium-feature.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import { useFreeTrialAvailable, useIsFreeApp, useNavigateToPricingPage } from '../hooks/instance';
+
+interface PremiumFeatureProps {
+    children: React.ReactNode;
+}
+
+export function PremiumFeature({ children }: PremiumFeatureProps) {
+    const isFreeApp = useIsFreeApp();
+    const isFreeTrialAvailable = useFreeTrialAvailable();
+    const navigateToPricingPage = useNavigateToPricingPage();
```
[the remark this answers is not in the document]
Comment (denysdovhan) [reply custom-products-catalog/template/src/dashboard/components/premium-feature.tsx:12]: Good point. Should be easy to fix.

File: custom-products-catalog/template/src/dashboard/components/premium-feature.tsx (hunk 2 above)
Comment (denysdovhan) [reply custom-products-catalog/template/src/dashboard/components/premium-feature.tsx:18]: Good catch. I missed that.

Maybe we need to think how we can add a linter to our templates. It's really easy to miss this kind of mistakes.

File: custom-products-catalog/template/src/dashboard/pages/page.tsx (hunk 3 above)
Comment (denysdovhan) [reply custom-products-catalog/template/src/dashboard/pages/page.tsx:172]: My initial though was that `isFree` should indicate when user's have an active Premium subscription and disable the button when subscription is not purchased. This should happen regardless of free trial availability.

File: custom-products-catalog/template/src/dashboard/pages/page.tsx (hunk 1 above)
Comment (denysdovhan) [reply custom-products-catalog/template/src/dashboard/pages/page.tsx:171]: I will investigate this in a separate PR. I can open a Jira for that, so we don't forget.

**Comment by denysdovhan** [comment]
@AdirAmsalem fixed. Please review again 👀

**Review by AdirAmsalem — APPROVED** [review]

**Review by AdirAmsalem — APPROVED** [review]

**Comment by denysdovhan** [comment]
Just checking once again that everything is being generated and works as expected...
I will merge this PR as soon as I verify that.
